packages/syntax-core/src/Modal/Modal.tsx

```diff
@@ -0,0 +1,160 @@
+import { type ReactElement } from "react";
+import Heading from "../Heading/Heading";
+import Box from "../Box/Box";
+
+import FocusTrap from "./FocusTrap";
+import StopScroll from "./StopScroll";
+import Layer from "./Layer";
+import styles from "./Modal.module.css";
+
+function XIcon({ color = "#000" }: { color?: string }) {
+  return (
+    <svg xmlns="http://www.w3.org/2000/svg" width="12" height="12" fill={color}>
+      <path
+        fill="inherit"
+        d="M11.25.758a.83.83 0 0 0-1.175 0L6 4.825 1.925.75A.83.83 0 1 0 .75 1.925L4.825 6 .75 10.075a.83.83 0 1 0 1.175 1.175L6 7.175l4.075 4.075a.83.83 0 1 0 1.175-1.175L7.175 6l4.075-4.075a.835.835 0 0 0 0-1.167Z"
+      />
+    </svg>
+  );
+}
+
+// Note: Only sm + lg size currently. design thinks there should only be two sizes.
+// If there IS a md size at some point, we should use the "size" const.
+const ModalSizes = ["sm", "lg"] as const;
+
+type ModalType = {
+  /**
+   * The children inside for the inside of a Modal.
+   */
+  children: JSX.Element;
+  /**
+   * The header inside a modal as a string.
+   */
+  header: string;
+  /**
+   * What the X button (or clicking out of the modal area) does.
+   * Typically used for closing the Modal
+   */
+  onDismiss: () => void;
+  /**
+   * The optional image rendered above the Modal.
+   * Image size should be 400w x 200h.
+   * If images aren't that sized, should ask design to give another image.
+   */
+  image?: JSX.Element;
+  /**
+   * The footer for the bottom area of the Modal.
+   * Typically used for rendering buttons.
+   */
+  footer?: JSX.Element;
+  /**
+   * The size of the card
+   *
+   * `sm`: 400px
```

Comment:
@somethiiing Could we make these bullet points? Just like we do in other places in Syntax

https://github.com/Cambly/syntax/blob/7c1f90584c4adb7e771cfa4899b85876d245ec4a/packages/syntax-core/src/IconButton/IconButton.tsx#L28-L30